packages/admin/src/implementation.ts

```diff
@@ -0,0 +1,233 @@
+  airnodeId: string,
+  requesterIndex: BigNumberish,
+  destination: string
+) {
+  const designatedWalletAddress = await deriveDesignatedWallet(airnodeRrp, airnodeId, requesterIndex);
+  await airnodeRrp.requestWithdrawal(airnodeId, requesterIndex, designatedWalletAddress, destination);
+  const filter = airnodeRrp.filters.WithdrawalRequested(null, null, null, null, null);
+
+  return new Promise<string>((resolve) =>
+    airnodeRrp.once(filter, (_, __, withdrawalRequestId) => {
+      resolve(withdrawalRequestId);
+    })
+  );
+}
+
+export async function checkWithdrawalRequest(airnodeRrp: AirnodeRrp, requestId: string) {
+  const filter = airnodeRrp.filters.WithdrawalFulfilled(null, null, requestId, null, null, null);
+
+  const logs = await airnodeRrp.queryFilter(filter);
+  if (logs.length === 0) {
+    return undefined;
+  }
+
+  const ethersLogParams = logs[0].args;
+  // remove array parameters from ethers response
+  const { airnodeId, amount, designatedWallet, destination, requesterIndex, withdrawalRequestId } = ethersLogParams;
+  const logParams = { airnodeId, amount, designatedWallet, destination, requesterIndex, withdrawalRequestId };
+  assertAllParamsAreReturned(logParams, ethersLogParams);
+
+  // cast ethers BigNumber for portability
+  return { ...logParams, amount: amount.toString(), requesterIndex: requesterIndex.toString() };
+}
+
+const isEthersWallet = (signer: any): signer is ethers.Wallet => !!signer.mnemonic;
+
+export async function setAirnodeParameters(airnodeRrp: AirnodeRrp, airnodeAdmin: string, authorizers: string[]) {
+  const wallet = airnodeRrp.signer;
+
+  if (!isEthersWallet(wallet)) {
+    throw new Error('Expected AirnodeRrp contract signer must be ethers.Wallet!');
+  }
+
+  const hdNode = ethers.utils.HDNode.fromMnemonic(wallet.mnemonic.phrase);
+  const xpub = hdNode.neuter().extendedKey;
+  const masterWallet = ethers.Wallet.fromMnemonic(wallet.mnemonic.phrase, 'm').connect(
+    airnodeRrp.provider as ethers.providers.Provider
+  );
+  // Assuming masterWallet has funds to make the transaction below
+  await airnodeRrp.connect(masterWallet).setAirnodeParametersAndForwardFunds(airnodeAdmin, xpub, authorizers);
+  const filter = airnodeRrp.filters.AirnodeParametersSet(null, null, null, null);
+
+  return new Promise<string>((resolve) =>
+    airnodeRrp.once(filter, (airnodeId) => {
+      resolve(airnodeId);
+    })
+  );
+}
+
+export async function deriveEndpointId(oisTitle: string, endpointName: string) {
+  return ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['string'], [`${oisTitle}_${endpointName}`]));
+}
+
+export async function clientAddressToNoRequests(airnodeRrp: AirnodeRrp, clientAddress: string) {
+  return (await airnodeRrp.clientAddressToNoRequests(clientAddress)).toString();
+}
+
+export async function getAirnodeParameters(airnodeRrp: AirnodeRrp, airnodeId: string) {
+  const ethersParams = await airnodeRrp.getAirnodeParametersAndBlockNumber(airnodeId);
```

Comment:
Not really important but it would make more sense to use `getAirnodeParameters()`

Reply:
Yeah, I didn't want to support both of these functions so I implemented the more generic one - but picked the shorter name for simplicity.